open-metadata-resources/open-metadata-deployment/docker/ranger/Dockerfile

```diff
@@ -66,109 +30,67 @@ RUN mvn -Pall -DskipTests=true clean compile package install assembly:assembly
-
-WORKDIR /opt
-RUN gunzip -c /root/ranger/target/ranger-${build}-admin.tar.gz | tar -xvf -
-
-RUN rm -fr /root/ranger/target
-# Get the mysql connector ready
-
-# Setup solr - needs java, contributed script for local solr install
-#
-# Script#s default port is 6083. 
-WORKDIR /opt/ranger-${build}-admin/contrib/solr_for_audit_setup
+# antrun                                      ranger-2.2.0-migration-util.tar.gz
+# archive-tmp                                 ranger-2.2.0-migration-util.zip
+# maven-shared-archive-resources              ranger-2.2.0-ranger-tools.tar.gz
+# ranger-2.2.0-admin.tar.gz          ranger-2.2.0-ranger-tools.zip
+# ranger-2.2.0-admin.zip             ranger-2.2.0-solr-plugin.tar.gz
+# ranger-2.2.0-atlas-plugin.tar.gz   ranger-2.2.0-solr-plugin.zip
+# ranger-2.2.0-atlas-plugin.zip      ranger-2.2.0-sqoop-plugin.tar.gz
+# ranger-2.2.0-hbase-plugin.tar.gz   ranger-2.2.0-sqoop-plugin.zip
+# ranger-2.2.0-hbase-plugin.zip      ranger-2.2.0-src.tar.gz
+# ranger-2.2.0-hdfs-plugin.tar.gz    ranger-2.2.0-src.zip
+# ranger-2.2.0-hdfs-plugin.zip       ranger-2.2.0-storm-plugin.tar.gz
+# ranger-2.2.0-hive-plugin.tar.gz    ranger-2.2.0-storm-plugin.zip
+# ranger-2.2.0-hive-plugin.zip       ranger-2.2.0-tagsync.tar.gz
+# ranger-2.2.0-kafka-plugin.tar.gz   ranger-2.2.0-tagsync.zip
+# ranger-2.2.0-kafka-plugin.zip      ranger-2.2.0-usersync.tar.gz
+# ranger-2.2.0-kms.tar.gz            ranger-2.2.0-usersync.zip
+# ranger-2.2.0-kms.zip               ranger-2.2.0-yarn-plugin.tar.gz
+# ranger-2.2.0-knox-plugin.tar.gz    ranger-2.2.0-yarn-plugin.zip
+# ranger-2.2.0-knox-plugin.zip       rat.txt
+# ranger-2.2.0-kylin-plugin.tar.gz   version
+# ranger-2.2.0-kylin-plugin.zip
+
+# Just going with the admin server here -- to include others, extract to a unique location and then copy path across
+# to runtime image below
+RUN mkdir -p /opt/ranger-admin
+RUN tar xzf /root/ranger/target/ranger-*-admin.tar.gz --strip-components 1 -C /opt/ranger-admin
+
+COPY dist/ranger-docker-start.sh /opt/ranger-admin/ranger-docker-start.sh
+RUN chmod 755 /opt/ranger-admin/ranger-docker-start.sh
+
+# Configure and install Solr locally
+WORKDIR /opt/ranger-admin/contrib/solr_for_audit_setup
 RUN sed -i 's|SOLR_INSTALL=false|SOLR_INSTALL=true|' install.properties
-RUN sed -i "s|SOLR_DOWNLOAD_URL=|SOLR_DOWNLOAD_URL=http://www.mirrorservice.org/sites/ftp.apache.org/lucene/solr/${solrver}/solr-${solrver}.tgz|" install.properties
-
-#Update path to include java, so that when we run the container it is available
-#default shell isn't bash :-(
-
-#RUN echo "export JAVA_HOME=/usr/lib/jvm/default-java" >> /root/.bashrc
-RUN echo "export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64" >> /root/.profile
-RUN echo "export PATH=${JAVA_HOME}/bin:$PATH" >> /root/.profile
-RUN chmod +x /root/.profile
+RUN sed -i "s|SOLR_DOWNLOAD_URL=|SOLR_DOWNLOAD_URL=http://www.mirrorservice.org/sites/ftp.apache.org/lucene/solr/7.7.1/solr-7.7.1.tgz|" install.properties
```

Comment:
the solr version can be a little fragile - usually the last point release is left. Just a note.